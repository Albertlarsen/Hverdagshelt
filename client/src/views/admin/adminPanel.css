
.panels {

}


.nyPanel{
    padding-top: 1em;
    height: 15em;
    background-color: #ffffff;
    border: solid 1px;
    border-color: #eaeaea;
    box-shadow: #eaefef 1px 1px 10px;
}


.nyPanel a {
    padding: 6px 8px 6px 16px;
    text-decoration: none;
    font-family: "Raleway", "HelveticaNeue", "Helvetica Neue", Helvetica, Arial, sans-serif;
    font-size: 15px;
}


.nyPanel .buttons {
    padding-left: 1em;
    padding-right: 1em;
    padding-top: 1em;
}


.nyPanel p {
    font-weight: bold;
    color: #3e4051;
}

.nyPanel hr {
    padding: 0;
    margin: 0;
}

.buttons a{
}


#col {
    padding-top: 4em;
}

.title {
    background-color: blue;
}