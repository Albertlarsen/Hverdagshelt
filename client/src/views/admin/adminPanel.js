import React, { Component } from 'react';
import { Grid, Row, Col, ListGroup, ListGroupItem, Table, Image, Panel, Button, ButtonGroup } from 'react-bootstrap';
import css from './adminPanel.css';

// get viktige hendelser


export class adminPanel extends Component {

  state = {};

  componentDidMount() {
  };



  render() {
    let buttonStyle = "primary";
    return (
      <div className="panels">

        <Grid>

          <Col xs={12} md={4} lg={4} id="col">
            <div className="nyPanel">
              <div >
              <Row align="center"><p>Kommuner & Kategorier</p></Row>
              <hr/>
              </div>
              <div className="buttons" >
                <Button bsStyle={buttonStyle} href="/#/addCategory" block><span className="glyphicon glyphicon-plus"/> Lag en kategori</Button>
                <Button bsStyle={buttonStyle} href="/#/" block><span className="glyphicon glyphicon-edit"/> Administrere Kategorier</Button>
                <Button bsStyle={buttonStyle} href="/#/" block><span className="glyphicon glyphicon-edit"/> Administrere Kommuner</Button>
              </div>
            </div>
          </Col>

          <Col xs={12} md={4} lg={4} id="col">
            <div className="nyPanel">
              <Row align="center"><p>Saker & Hendelser</p></Row>
              <hr/>
              <div className="buttons">
              <Button bsStyle={buttonStyle} href="/#/" block><span className="glyphicon glyphicon-edit"/> Administrere Hendelser</Button>
              <Button bsStyle={buttonStyle} href="/#/" block><span className="glyphicon glyphicon-edit"/> Administrere Saker</Button>
              </div>
            </div>
          </Col>

          <Col xs={12} md={4} lg={4} id="col">
            <div className="nyPanel">
              <Row align="center"><p>Brukere</p></Row>
              <hr/>
              <div className="buttons">
              <Button bsStyle={buttonStyle} href="/#/" block><span className="glyphicon glyphicon-plus"/> Lag en ansatt</Button>
              <Button bsStyle={buttonStyle} href="/#/admin/sendMailTilBruker" block><span className="glyphicon glyphicon-envelope"/> Kontakt bruker</Button>
              <Button bsStyle={buttonStyle} href="/#/" block><span className="glyphicon glyphicon-edit"/> Administrere Brukere</Button>
             </div>
            </div>
          </Col>


        </Grid>


      </div>
    );
  }
}