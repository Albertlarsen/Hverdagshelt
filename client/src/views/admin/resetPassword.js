//@flow
import { MailService } from '../../services';
import { Component } from 'react';
import * as React from 'react';
import ReactDOM from 'react-dom';
import { Button, Form, FormControl, FormGroup, Grid, Col } from 'react-bootstrap';
import resetPassword from './resetPassword.css';

let mailService = new MailService();

export class AdminResetUserPassword extends Component<Props, State> {
  constructor(props) {
    super(props);
    this.state = {
      email: { to: '' }
    };
  }

  handleStringChange = () => (event: SyntheticInputEvent<HTMLInputElement>) => {
    this.setState({
      email: { to: event.target.value }
    });
  };

  render() {
    return (
      <div>
        <Grid>

          <Col xs={2} sm={2} md={2} lg={2}></Col>

          <Col xs={8} sm={8} md={8} lg={8}>
            <h2>Tilbakestill en brukers passord</h2>
            <p>Brukeren vil få tilsendt et tilfeldig generert passord på mail</p>

            <Form>
              <FormGroup validationState={this.getValidationStateEmail()}>
                <div className="textfield">
                  <FormControl
                    value={this.state.email.to}
                    onChange={this.handleStringChange()}
                    placeholder="Epost"
                  />
                  <FormControl.Feedback/>
                </div>
              </FormGroup>
              <Button bsStyle="primary" onClick={this.resetPassword}>
                Tilbakestill passord
              </Button>
            </Form>
          </Col>

          <Col xs={2} sm={2} md={2} lg={2}></Col>
        </Grid>
      </div>
    );
  }

  getValidationStateEmail() {
    var validator = require('email-validator');
    const length = this.state.email.to.length;
    console.log(this.state.email.to.length);
    const bool = validator.validate(this.state.email.to);
    if (length == 0) return;
    else if (!bool) return 'warning';
    else if (bool) return 'success';
  }

  resetPassword = () => {
    console.log(this.state.email.to);
    mailService.sendResetPasswordMail({ to: this.state.email.to });
  };
}
