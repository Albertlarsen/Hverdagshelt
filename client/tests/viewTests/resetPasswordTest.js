// @flow

import {shallow} from "enzyme/build";
import * as React from "react";
import {AdminResetUserPassword} from "../../src/views/admin/resetPassword";

describe('Test for AdminResetUserPassword view', () => {
    const wrapper = shallow(<AdminResetUserPassword />);

    it('renders correctly', () => {
        expect(wrapper).toMatchSnapshot();
    });
});