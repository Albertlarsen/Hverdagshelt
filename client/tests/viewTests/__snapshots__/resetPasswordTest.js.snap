// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test for AdminResetUserPassword view renders correctly 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <AdminResetUserPassword />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": <Grid
        bsClass="container"
        componentClass="div"
        fluid={false}
      >
        <Col
          bsClass="col"
          componentClass="div"
          lg={2}
          md={2}
          sm={2}
          xs={2}
        />
        <Col
          bsClass="col"
          componentClass="div"
          lg={8}
          md={8}
          sm={8}
          xs={8}
        >
          <h2>
            Tilbakestill en brukers passord
          </h2>
          <p>
            Brukeren vil få tilsendt et tilfeldig generert passord på mail
          </p>
          <Form
            bsClass="form"
            componentClass="form"
            horizontal={false}
            inline={false}
          >
            <FormGroup
              bsClass="form-group"
            >
              <div
                className="textfield"
              >
                <FormControl
                  bsClass="form-control"
                  componentClass="input"
                  onChange={[Function]}
                  placeholder="Epost"
                  value=""
                />
                <FormControlFeedback
                  bsClass="form-control-feedback"
                  bsRole="feedback"
                />
              </div>
            </FormGroup>
            <Button
              active={false}
              block={false}
              bsClass="btn"
              bsStyle="primary"
              disabled={false}
              onClick={[Function]}
            >
              Tilbakestill passord
            </Button>
          </Form>
        </Col>
        <Col
          bsClass="col"
          componentClass="div"
          lg={2}
          md={2}
          sm={2}
          xs={2}
        />
      </Grid>,
    },
    "ref": null,
    "rendered": Object {
      "instance": null,
      "key": undefined,
      "nodeType": "class",
      "props": Object {
        "bsClass": "container",
        "children": Array [
          <Col
            bsClass="col"
            componentClass="div"
            lg={2}
            md={2}
            sm={2}
            xs={2}
          />,
          <Col
            bsClass="col"
            componentClass="div"
            lg={8}
            md={8}
            sm={8}
            xs={8}
          >
            <h2>
              Tilbakestill en brukers passord
            </h2>
            <p>
              Brukeren vil få tilsendt et tilfeldig generert passord på mail
            </p>
            <Form
              bsClass="form"
              componentClass="form"
              horizontal={false}
              inline={false}
            >
              <FormGroup
                bsClass="form-group"
              >
                <div
                  className="textfield"
                >
                  <FormControl
                    bsClass="form-control"
                    componentClass="input"
                    onChange={[Function]}
                    placeholder="Epost"
                    value=""
                  />
                  <FormControlFeedback
                    bsClass="form-control-feedback"
                    bsRole="feedback"
                  />
                </div>
              </FormGroup>
              <Button
                active={false}
                block={false}
                bsClass="btn"
                bsStyle="primary"
                disabled={false}
                onClick={[Function]}
              >
                Tilbakestill passord
              </Button>
            </Form>
          </Col>,
          <Col
            bsClass="col"
            componentClass="div"
            lg={2}
            md={2}
            sm={2}
            xs={2}
          />,
        ],
        "componentClass": "div",
        "fluid": false,
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "bsClass": "col",
            "componentClass": "div",
            "lg": 2,
            "md": 2,
            "sm": 2,
            "xs": 2,
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "bsClass": "col",
            "children": Array [
              <h2>
                Tilbakestill en brukers passord
              </h2>,
              <p>
                Brukeren vil få tilsendt et tilfeldig generert passord på mail
              </p>,
              <Form
                bsClass="form"
                componentClass="form"
                horizontal={false}
                inline={false}
              >
                <FormGroup
                  bsClass="form-group"
                >
                  <div
                    className="textfield"
                  >
                    <FormControl
                      bsClass="form-control"
                      componentClass="input"
                      onChange={[Function]}
                      placeholder="Epost"
                      value=""
                    />
                    <FormControlFeedback
                      bsClass="form-control-feedback"
                      bsRole="feedback"
                    />
                  </div>
                </FormGroup>
                <Button
                  active={false}
                  block={false}
                  bsClass="btn"
                  bsStyle="primary"
                  disabled={false}
                  onClick={[Function]}
                >
                  Tilbakestill passord
                </Button>
              </Form>,
            ],
            "componentClass": "div",
            "lg": 8,
            "md": 8,
            "sm": 8,
            "xs": 8,
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": "Tilbakestill en brukers passord",
              },
              "ref": null,
              "rendered": "Tilbakestill en brukers passord",
              "type": "h2",
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": "Brukeren vil få tilsendt et tilfeldig generert passord på mail",
              },
              "ref": null,
              "rendered": "Brukeren vil få tilsendt et tilfeldig generert passord på mail",
              "type": "p",
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "class",
              "props": Object {
                "bsClass": "form",
                "children": Array [
                  <FormGroup
                    bsClass="form-group"
                  >
                    <div
                      className="textfield"
                    >
                      <FormControl
                        bsClass="form-control"
                        componentClass="input"
                        onChange={[Function]}
                        placeholder="Epost"
                        value=""
                      />
                      <FormControlFeedback
                        bsClass="form-control-feedback"
                        bsRole="feedback"
                      />
                    </div>
                  </FormGroup>,
                  <Button
                    active={false}
                    block={false}
                    bsClass="btn"
                    bsStyle="primary"
                    disabled={false}
                    onClick={[Function]}
                  >
                    Tilbakestill passord
                  </Button>,
                ],
                "componentClass": "form",
                "horizontal": false,
                "inline": false,
              },
              "ref": null,
              "rendered": Array [
                Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "class",
                  "props": Object {
                    "bsClass": "form-group",
                    "children": <div
                      className="textfield"
                    >
                      <FormControl
                        bsClass="form-control"
                        componentClass="input"
                        onChange={[Function]}
                        placeholder="Epost"
                        value=""
                      />
                      <FormControlFeedback
                        bsClass="form-control-feedback"
                        bsRole="feedback"
                      />
                    </div>,
                    "validationState": undefined,
                  },
                  "ref": null,
                  "rendered": Object {
                    "instance": null,
                    "key": undefined,
                    "nodeType": "host",
                    "props": Object {
                      "children": Array [
                        <FormControl
                          bsClass="form-control"
                          componentClass="input"
                          onChange={[Function]}
                          placeholder="Epost"
                          value=""
                        />,
                        <FormControlFeedback
                          bsClass="form-control-feedback"
                          bsRole="feedback"
                        />,
                      ],
                      "className": "textfield",
                    },
                    "ref": null,
                    "rendered": Array [
                      Object {
                        "instance": null,
                        "key": undefined,
                        "nodeType": "class",
                        "props": Object {
                          "bsClass": "form-control",
                          "componentClass": "input",
                          "onChange": [Function],
                          "placeholder": "Epost",
                          "value": "",
                        },
                        "ref": null,
                        "rendered": null,
                        "type": [Function],
                      },
                      Object {
                        "instance": null,
                        "key": undefined,
                        "nodeType": "class",
                        "props": Object {
                          "bsClass": "form-control-feedback",
                          "bsRole": "feedback",
                        },
                        "ref": null,
                        "rendered": null,
                        "type": [Function],
                      },
                    ],
                    "type": "div",
                  },
                  "type": [Function],
                },
                Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "class",
                  "props": Object {
                    "active": false,
                    "block": false,
                    "bsClass": "btn",
                    "bsStyle": "primary",
                    "children": "Tilbakestill passord",
                    "disabled": false,
                    "onClick": [Function],
                  },
                  "ref": null,
                  "rendered": "Tilbakestill passord",
                  "type": [Function],
                },
              ],
              "type": [Function],
            },
          ],
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "bsClass": "col",
            "componentClass": "div",
            "lg": 2,
            "md": 2,
            "sm": 2,
            "xs": 2,
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
      ],
      "type": [Function],
    },
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": <Grid
          bsClass="container"
          componentClass="div"
          fluid={false}
        >
          <Col
            bsClass="col"
            componentClass="div"
            lg={2}
            md={2}
            sm={2}
            xs={2}
          />
          <Col
            bsClass="col"
            componentClass="div"
            lg={8}
            md={8}
            sm={8}
            xs={8}
          >
            <h2>
              Tilbakestill en brukers passord
            </h2>
            <p>
              Brukeren vil få tilsendt et tilfeldig generert passord på mail
            </p>
            <Form
              bsClass="form"
              componentClass="form"
              horizontal={false}
              inline={false}
            >
              <FormGroup
                bsClass="form-group"
              >
                <div
                  className="textfield"
                >
                  <FormControl
                    bsClass="form-control"
                    componentClass="input"
                    onChange={[Function]}
                    placeholder="Epost"
                    value=""
                  />
                  <FormControlFeedback
                    bsClass="form-control-feedback"
                    bsRole="feedback"
                  />
                </div>
              </FormGroup>
              <Button
                active={false}
                block={false}
                bsClass="btn"
                bsStyle="primary"
                disabled={false}
                onClick={[Function]}
              >
                Tilbakestill passord
              </Button>
            </Form>
          </Col>
          <Col
            bsClass="col"
            componentClass="div"
            lg={2}
            md={2}
            sm={2}
            xs={2}
          />
        </Grid>,
      },
      "ref": null,
      "rendered": Object {
        "instance": null,
        "key": undefined,
        "nodeType": "class",
        "props": Object {
          "bsClass": "container",
          "children": Array [
            <Col
              bsClass="col"
              componentClass="div"
              lg={2}
              md={2}
              sm={2}
              xs={2}
            />,
            <Col
              bsClass="col"
              componentClass="div"
              lg={8}
              md={8}
              sm={8}
              xs={8}
            >
              <h2>
                Tilbakestill en brukers passord
              </h2>
              <p>
                Brukeren vil få tilsendt et tilfeldig generert passord på mail
              </p>
              <Form
                bsClass="form"
                componentClass="form"
                horizontal={false}
                inline={false}
              >
                <FormGroup
                  bsClass="form-group"
                >
                  <div
                    className="textfield"
                  >
                    <FormControl
                      bsClass="form-control"
                      componentClass="input"
                      onChange={[Function]}
                      placeholder="Epost"
                      value=""
                    />
                    <FormControlFeedback
                      bsClass="form-control-feedback"
                      bsRole="feedback"
                    />
                  </div>
                </FormGroup>
                <Button
                  active={false}
                  block={false}
                  bsClass="btn"
                  bsStyle="primary"
                  disabled={false}
                  onClick={[Function]}
                >
                  Tilbakestill passord
                </Button>
              </Form>
            </Col>,
            <Col
              bsClass="col"
              componentClass="div"
              lg={2}
              md={2}
              sm={2}
              xs={2}
            />,
          ],
          "componentClass": "div",
          "fluid": false,
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "class",
            "props": Object {
              "bsClass": "col",
              "componentClass": "div",
              "lg": 2,
              "md": 2,
              "sm": 2,
              "xs": 2,
            },
            "ref": null,
            "rendered": null,
            "type": [Function],
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "class",
            "props": Object {
              "bsClass": "col",
              "children": Array [
                <h2>
                  Tilbakestill en brukers passord
                </h2>,
                <p>
                  Brukeren vil få tilsendt et tilfeldig generert passord på mail
                </p>,
                <Form
                  bsClass="form"
                  componentClass="form"
                  horizontal={false}
                  inline={false}
                >
                  <FormGroup
                    bsClass="form-group"
                  >
                    <div
                      className="textfield"
                    >
                      <FormControl
                        bsClass="form-control"
                        componentClass="input"
                        onChange={[Function]}
                        placeholder="Epost"
                        value=""
                      />
                      <FormControlFeedback
                        bsClass="form-control-feedback"
                        bsRole="feedback"
                      />
                    </div>
                  </FormGroup>
                  <Button
                    active={false}
                    block={false}
                    bsClass="btn"
                    bsStyle="primary"
                    disabled={false}
                    onClick={[Function]}
                  >
                    Tilbakestill passord
                  </Button>
                </Form>,
              ],
              "componentClass": "div",
              "lg": 8,
              "md": 8,
              "sm": 8,
              "xs": 8,
            },
            "ref": null,
            "rendered": Array [
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "children": "Tilbakestill en brukers passord",
                },
                "ref": null,
                "rendered": "Tilbakestill en brukers passord",
                "type": "h2",
              },
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "children": "Brukeren vil få tilsendt et tilfeldig generert passord på mail",
                },
                "ref": null,
                "rendered": "Brukeren vil få tilsendt et tilfeldig generert passord på mail",
                "type": "p",
              },
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "class",
                "props": Object {
                  "bsClass": "form",
                  "children": Array [
                    <FormGroup
                      bsClass="form-group"
                    >
                      <div
                        className="textfield"
                      >
                        <FormControl
                          bsClass="form-control"
                          componentClass="input"
                          onChange={[Function]}
                          placeholder="Epost"
                          value=""
                        />
                        <FormControlFeedback
                          bsClass="form-control-feedback"
                          bsRole="feedback"
                        />
                      </div>
                    </FormGroup>,
                    <Button
                      active={false}
                      block={false}
                      bsClass="btn"
                      bsStyle="primary"
                      disabled={false}
                      onClick={[Function]}
                    >
                      Tilbakestill passord
                    </Button>,
                  ],
                  "componentClass": "form",
                  "horizontal": false,
                  "inline": false,
                },
                "ref": null,
                "rendered": Array [
                  Object {
                    "instance": null,
                    "key": undefined,
                    "nodeType": "class",
                    "props": Object {
                      "bsClass": "form-group",
                      "children": <div
                        className="textfield"
                      >
                        <FormControl
                          bsClass="form-control"
                          componentClass="input"
                          onChange={[Function]}
                          placeholder="Epost"
                          value=""
                        />
                        <FormControlFeedback
                          bsClass="form-control-feedback"
                          bsRole="feedback"
                        />
                      </div>,
                      "validationState": undefined,
                    },
                    "ref": null,
                    "rendered": Object {
                      "instance": null,
                      "key": undefined,
                      "nodeType": "host",
                      "props": Object {
                        "children": Array [
                          <FormControl
                            bsClass="form-control"
                            componentClass="input"
                            onChange={[Function]}
                            placeholder="Epost"
                            value=""
                          />,
                          <FormControlFeedback
                            bsClass="form-control-feedback"
                            bsRole="feedback"
                          />,
                        ],
                        "className": "textfield",
                      },
                      "ref": null,
                      "rendered": Array [
                        Object {
                          "instance": null,
                          "key": undefined,
                          "nodeType": "class",
                          "props": Object {
                            "bsClass": "form-control",
                            "componentClass": "input",
                            "onChange": [Function],
                            "placeholder": "Epost",
                            "value": "",
                          },
                          "ref": null,
                          "rendered": null,
                          "type": [Function],
                        },
                        Object {
                          "instance": null,
                          "key": undefined,
                          "nodeType": "class",
                          "props": Object {
                            "bsClass": "form-control-feedback",
                            "bsRole": "feedback",
                          },
                          "ref": null,
                          "rendered": null,
                          "type": [Function],
                        },
                      ],
                      "type": "div",
                    },
                    "type": [Function],
                  },
                  Object {
                    "instance": null,
                    "key": undefined,
                    "nodeType": "class",
                    "props": Object {
                      "active": false,
                      "block": false,
                      "bsClass": "btn",
                      "bsStyle": "primary",
                      "children": "Tilbakestill passord",
                      "disabled": false,
                      "onClick": [Function],
                    },
                    "ref": null,
                    "rendered": "Tilbakestill passord",
                    "type": [Function],
                  },
                ],
                "type": [Function],
              },
            ],
            "type": [Function],
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "class",
            "props": Object {
              "bsClass": "col",
              "componentClass": "div",
              "lg": 2,
              "md": 2,
              "sm": 2,
              "xs": 2,
            },
            "ref": null,
            "rendered": null,
            "type": [Function],
          },
        ],
        "type": [Function],
      },
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getDerivedStateFromProps": true,
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
  },
}
`;
